src/Simulation/Core/TypeExtensions.cs

```diff
@@ -144,6 +146,38 @@ public static Type[] GetTupleFieldTypes(this Type arg)
             }
         }
 
+        /// <summary>
+        /// Given a <see cref="Type"/> and its arguments, format its non-qubit arguments into a string.
+        /// Returns null if no arguments found.
+        /// </summary>
+        public static string? ArgsToString(this Type t, object args)
+        {
+            var argsStrings = t.GetFields()
+                .Select(f =>
+                {
+                    var argString = null as string;
+
+                    // If field is a tuple, recursively extract its inner arguments and format as a tuple string.
+                    if (f.FieldType.IsTuple())
```

Comment:
what about arrays? does this work well with them?

Reply:
Good catch, I don't think so. I'll add tests for this function to make sure it works with arrays.